qubesusbproxy/core3ext.py

```diff
@@ -389,9 +608,9 @@ async def on_device_detach_usb(self, vm, event, device):
     @qubes.ext.handler('domain-start')
     async def on_domain_start(self, vm, _event, **_kwargs):
         # pylint: disable=unused-argument
-        for assignment in vm.devices['usb'].assignments(persistent=True):
-            device = assignment.device
-            await self.on_device_attach_usb(vm, '', device, options={})
+        for assignment in vm.devices['usb'].get_assigned_devices():
+            asyncio.ensure_future(self.attach_and_notify(
```

Comment:
I think this should be `await` instead of `asyncio.ensure_future` and that's what the test detected (see the "Task exception was never retrieved" part). And the actual attach probably failed because test already started cleaning up (but that's just my guess)...

Reply:
youre right